tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -1671,175 +1671,215 @@ TEST_F(GPUTTensorCoreTest, FusionAmpereViewMatmulTN_CUDA) {
+    mma_utils::scheduleWarpTileWithNoReduction(tv2, gemm_tile);
+    //           -8   -7 -6 -5 -4 -3 -2 -1
+    // [Mo No Ko Kwo Mwo Nwo Mw Nw Mi Ni Ki]
+    tv0cr->computeAt(tv2c, -4);
+    tv1cr->computeAt(tv2c, -4);
+
+    // Schedule gmem read and smem write:
+    //
+    // [Mo,Ko,M,K]
+    // Swizzle tv0: 128 x 32 tile:
+    tv0cw->split(-2, 8);
+    tv0cw->split(-2, 2);
+    tv0cw->split(-1, 8);
+    //        -5   -4 -3 -2 -1
+    // [Mo,Ko,Mo16,M4,M2,Ko4,K8]
+    tv0cw->swizzle(Swizzle2DType::XOR, -4, -2);
+    tv0cw->merge(-4);
+    tv0cw->merge(-3);
+    //         -3   -2  -1
+    // [Mo,Ko,Mo16,warp,K8]
+    tv0cw->split(-3, 4);
+    tv0cw->split(-3, 2);
+    //             -4  -3   -2  -1
+    // [Mo,Ko, S4, wz2, wy2, warp,K8]
+    tv0cw->axis(-4)->parallelize(ParallelType::TIDz);
+    tv0cw->axis(-3)->parallelize(ParallelType::TIDy);
+    tv0cw->axis(-2)->parallelize(ParallelType::TIDx);
+    tv0cw->axis(-1)->parallelize(ParallelType::Vectorize);
+
+    tv0cw->setMemoryType(MemoryType::Shared);
+    // [Mo,Ko,i,wy,wx,v]
+
+    // [No,Ko,N,K]
+    // Swizzle tv0: 128 x 32 tile:
+    tv1cw->split(-2, 8);
+    tv1cw->split(-2, 2);
+    tv1cw->split(-1, 8);
+    //        -5   -4 -3 -2 -1
+    // [No,Ko,No16,N4,N2,Ko4,K8]
+    tv1cw->swizzle(Swizzle2DType::XOR, -4, -2);
+    tv1cw->merge(-4);
+    tv1cw->merge(-3);
+    //         -3   -2  -1
+    // [No,Ko,No16,warp,K8]
+    tv1cw->split(-3, 4);
+    tv1cw->split(-3, 2);
+    //             -4  -3   -2  -1
+    // [No,Ko, S4, wz2, wy2, warp,K8]
+    tv1cw->axis(-4)->parallelize(ParallelType::TIDz);
+    tv1cw->axis(-3)->parallelize(ParallelType::TIDy);
+    tv1cw->axis(-2)->parallelize(ParallelType::TIDx);
+    tv1cw->axis(-1)->parallelize(ParallelType::Vectorize);
+
+    tv1cw->setMemoryType(MemoryType::Shared);
+    // Schedule mma input
+    tv0cr->applyMmaSwizzle(MmaOperand::A);
+
+    // [... Mi, Ni, Ki]
+    tv0b->reorder({{-2, -3}, {-3, -2}});
+    tv0b->applyMmaSwizzle(MmaOperand::A);
+
+    tv1cr->applyMmaSwizzle(MmaOperand::B);
+    tv1b->applyMmaSwizzle(MmaOperand::B);
+
+    // Schedule mma output
+    tv2c->applyMmaSwizzle(MmaOperand::Accumulator);
+    tv2->applyMmaSwizzle(MmaOperand::Accumulator);
+
+    // Parallelize
+    //  0   1  2  3   4   5  6   7  8  9  10
+    // [Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
+    tv2c->axis(4)->parallelize(ParallelType::TIDz);
+    tv2c->axis(5)->parallelize(ParallelType::TIDy);
+
+    // Parallelize
+    //  0  1  2   3   4   5  6  7
+    // [Mo No Mwo Nwo Mw Nw (Mi Ni)]
+    tv2->axis(0)->parallelize(ParallelType::BIDx);
+    tv2->axis(1)->parallelize(ParallelType::BIDy);
+    tv2->axis(2)->parallelize(ParallelType::TIDz);
+    tv2->axis(3)->parallelize(ParallelType::TIDy);
+
+    tv0cw->doubleBuffer();
+    tv1cw->doubleBuffer();
+    tv0cr->doubleBuffer();
+    tv1cr->doubleBuffer();
+
+    // Create input for either [M, K] @ [N, K] or [M, K] @ [K, N] (with strides)
```

Comment:
I think it's a bad habit to write `[M, K] @ [N, K]` as the `@` symbol really has no meaning other than mathematical matrix multiply. Instead you could write something like
> Create input for operand B having stride one in either the K or N dimension